### ISSUE TITLE
Unsolid walls

### ISSUE THREAD

**Issue by Pegwald** [aura-project/aura#176]
I have noticed that I am able to go places were I am not really supposed to. Like outside in Tir and inside dungeons. Creatures, mobs and even players are able to glitch through some places and I have no idea why. Here's some footage I recently recorded when I was just chilling and eating some pizza - https://www.youtube.com/watch?v=VTXkjRBuWDU

![how](https://cloud.githubusercontent.com/assets/13089450/12029427/1a882382-adeb-11e5-89c7-e723ee673681.PNG)

**Comment by Sairii** [comment]
Mobs glitching into stuff should be known.
But your dungeon video.... exec is gonna cry.
I have yet to do a dungeon on Aura so I didnt know xD

**Comment by Pegwald** [comment]
Poor exec

**Comment by exectails** [comment]
WTF happened in that video o_O"""

First of all, before some spreads false rumors again, we do have server side collision detection for monsters. It's currently deactivated for players though (#149).

Now, when we added dungeons, there was a bug in how the AI decided where to go. It was possible for the AI to detect a collision, walk into a different direction, and end up inside a prop. But that should be fixed, and in my tests back then it didn't happen anymore.

Due to players not being checked for collisions, it's possible to "glitch" through crannies, just like it was possible on officials in the past, which was, afaik, not fixed before C2 or C3. It's also currently possible for monster to spawn inside a prop, which could explain the sheep in the screen shot.

However, the video shows an entirely different problem, one that is completely unexplainable to me right now. You didn't trick the server, or glitch through a cranny, you downright walked through walls, which the *client* should *not allow*, unless your client is modified.

I'm not even sure this is an Aura problem... the *client* let you pass through walls. Granted, the server didn't stop you, but what was the *client* thinking...? Could you look up the parameters used for the dungeon in the logs? I mean... if it's a bug, it has to be reproducible.

@Tachiorz Have you ever seen something like this in your tests?

**Comment by Pegwald** [comment]
There were some walls I could not glitch through and I think those walls were facing a new wall on the other side. Maybe I have some modifications that allow me to click beyond - into the void, which really is a client issue. Right now I'm running Tiara with very few modifications, only to enable a data folder for the hotkey fix we were looking at earlier, exec.

There are also other places I can reach outside in tir that lets me walk on the mountain around Ciar and such, still accessable for other clients too (checked), but even then I was the only one able to go further into the void. I Believe this might have something to do with my client modifications after all...

**Comment by exectails** [comment]
Sounds like it, because the client really shouldn't let you do that =| Maybe you could try it with an unmodified client?

**Comment by Pegwald** [comment]
I'll give it a shot

**Comment by exectails** [comment]
Any more information on this?

**Comment by Pegwald** [comment]
Sorry for not responding in a while, but I have had issues with the client since removing the mods seemed to ruin it. If I should be able to check with a clean client, I need to either reinstall or borrow a pc with a clean install. I will consider doing the second option first before I go reinstall over again like i've done a hundred times already ^^ ''

**Comment by Pegwald** [comment]
I have some very bad news... Someone else on my server was able to do the same glitch as me, even at the same time as me when we were doing Rabbie Basic earlier today. That person does not have any mods installed.

**Comment by exectails** [comment]
>Could you look up the parameters used for the dungeon in the logs? I mean... if it's a bug, it has to be reproducible.

**Comment by Pegwald** [comment]
05.01.2016 12:09:31 [Debug] - Dungeon: Creating 'dunbarton_rabbie_low_dungeon', item id: 63110, seed: 1730785922, floorPlan: 1566937264

**Comment by exectails** [comment]
Huh. I can walk through walls, curious. On the first floor, right in the first statue room I was able to walk through the wall on the left, with the above parameters. Comparing the mini map and the output of Pale's DunGen, I can say that the floor layout is the same, even if it shouldn't cause this.

The problem is, that this makes 0 sense. The *client* should *not* let you walk through walls. No matter what the server tells it, the clients displays a wall, so from its POV there *is* a wall.

What's also weird, is that it doesn't let you walk through every wall, but only some, while it still lets you walk through all doors, as it's supposed to.

Here's the collision plan for the first floor, which is generated out of the prop shapes, that the client should be using as well. As you can see, there are solid walls between the start room (upper center) and the room on its left.

![dunbarton_rabbie_low_dungeon_10011](https://cloud.githubusercontent.com/assets/5593920/12116929/7c117dec-b3bd-11e5-92b0-91defacc6b36.png)

Tried that again, with random parameters. I was able to walk through the same wall in Rabbie Basic every time, the left wall of the start room, unless the door was there. In some instances I could walk through the north wall as well.

**Comment by exectails** [comment]
The only thing that comes to my mind is client bug o,o They have server-side collision detection for players now, so while the click behavior would be a little weird, it would prevent you from walking through those walls.

Though I don't remember anyone complaining about bugged walls in the past... they would've had to mess up the room prop or something after the collision detection for players was added.

**Comment by Pegwald** [comment]
Everything you mention sounds about right. Only some walls are "passable" meanwhile some are solid as supposed to. Keep in mind it could be all the dungeons since I've already experienced it in both Alby and Rabbie. I have not checked the other dungeons yet, but I dare say that it might pretty much look the same in those too.

**Comment by exectails** [comment]
Is there a normal dungeon where this happens? Ideally in the first room? I'd like to test the behavior on officials.

**Comment by exectails** [comment]
Apparently this is possible in Alby Normal as well =/ I've tested the behavior on NA, and the server is in fact resetting the player's position. It's possible to "glitch" into the wall for a fraction of a second, before the reset.

This is so weird... if the server has to reset the position, it means the *client* believes it can go through certain walls. But in the beginning there was no server side collision detection for players, and I don't remember this ever happening on officials, so they must have messed with the dungeons afterwards. However, the server still has the correct data, it's able to reset your position after all... how did they do this!?

Since this worked in 2 dungeons, in the first room, in the exact same spot, I'll assume it works in all of them. I guess they have a bug in their collision generator for client side dungeons, which makes parts of certain walls passable. I wonder why they've even messed with that code though, it's not like they've done anything with dungeons in years >_> I assume this hasn't been discovered yet, because nobody was actually able to walk through the walls, after the server got player collision detection.

Long story short, #149 should fix this, as it should make it impossible for players to pass/glitch through walls, no matter what the client says. I'll look into it soon-ish.

**Comment by exectails** [comment]
I've added player collision detection now, and I'm not able to pass through walls anymore in dungeons.
